docs/help/git_webhooks.md

```diff
@@ -0,0 +1,30 @@
+## Webhooks
+
+1. Generate a SSL certificate for the REST API. TODO: figure out the right
+   way to do that.
+
+Using acme.sh on the host already, I did TODO
+
+```
+$ACMESH --key-file $BITBOTDIR/ssl.key --cert-file $BITBOTDIR/ssl.cert --reloadcmd "$SYSTEMCTLRESTART bitbot"
+chmod -R 700 $BITBOTDIR/ssl.key
+chown -R bitbot:bitbot $BITBOTDIR
+```
+
+2. Enable the REST API in bot.conf by adjusting
+
+```
+# key/cert used for REST API
+tls-api-key              = /path/to/just/generated/key
+tls-api-certificate      = /path/to/just/generated/cert
+api-port                 = 5000
+```
+
+you may change the API port if you wish.
+
+3. restart the bot
+
+4. run `!config bot rest-api on`.
+
+5. Generate a random key for the git service to use, e.g. `pwgen 10`, pick
+   one and remember it. `/msg <bot> apikey <name> /api/github`
```

Comment:
I understand that this is a password protection so anyone cannot spam the `<host>:5000`. I presume `apikey` is a command, but is `<name>` the API key I just generated with `pwgen 10`? `/api/github` is probably what I will tell GitHub to call to, but I am setting up a Gitea webhook and I don't understand from the code would I just replace it with `/api/gitea`.

I also don't feel like trying to continue further as I have no idea if I can remove failed attempts or how to do that, and as I am unable to telnet or nmap the port, I don't know if my attempt would be going to fail anyway.

Reply:
`<name>` is just a label for the key - so you know who's sending in data by their API key. where'd you get `pwgen 10` from?

Reply:
I assumed it's a secret that only I and Git-someone knows.

So is the answer `/msg <bot> apikey <friendly-name> /api/gitea` ? Or does Gitea also use /api/github?

Reply:
ah, the apikey is autogenerated, nice